sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/branch/ProjectBranchesProvider.java

```diff
@@ -36,16 +34,19 @@
 
   private ProjectBranches branches = null;
 
-  public ProjectBranches provide(@Nullable ProjectBranchesLoader loader, ProjectKey projectKey, GlobalConfiguration settings) {
-    if (branches == null) {
-      if (loader == null || !settings.get(ScannerProperties.BRANCH_NAME).isPresent()) {
-        branches = new ProjectBranches(Collections.emptyList());
-      } else {
-        Profiler profiler = Profiler.create(LOG).startInfo(LOG_MSG);
-        branches = loader.load(projectKey.get());
-        profiler.stopInfo();
-      }
+  public ProjectBranches provide(@Nullable ProjectBranchesLoader loader, ProjectKey projectKey) {
+    if (branches != null) {
+      return branches;
     }
+
+    if (loader == null) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Change this condition so that it does not always evaluate to "false" ([squid:S2583](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2583))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fvsts_pr&issues=AWGU9dVgQeDOAQ7I7szE&open=AWGU9dVgQeDOAQ7I7szE)

[ISSUE_KEY]: AWGU9dVgQeDOAQ7I7szE